lsp/nls/src/linearization.rs

```diff
@@ -100,7 +288,7 @@ pub struct AnalysisHost {
     /// in their own scope immediately after the record, which
     /// gives the corresponding record field _term_ to the ident
     /// useable to construct a vale declaration.
-    record_fields: Option<(usize, Vec<Ident>)>,
+    record_fields: Option<(usize, Vec<(usize, Ident)>)>,
```

Comment:
It may be  useful to use a type alias for `usize`, to know what it means. I guess those are UIDs? It's not easy to know at first sight if this is supposed to be a length, an identifier, or something else. I am not saying it has to be done in this PR, but just as a general suggestion.

Reply:
Yes its Inearization item id's (which are indexes during the `Building` phase but thats a different matter)
I agree, we should have these be more clear throughout the module